diff --git a/node_modules/@gorhom/bottom-sheet/src/components/bottomSheetContainer/BottomSheetContainer.tsx b/node_modules/@gorhom/bottom-sheet/src/components/bottomSheetContainer/BottomSheetContainer.tsx
index 54eab00..f46f4c8 100644
--- a/node_modules/@gorhom/bottom-sheet/src/components/bottomSheetContainer/BottomSheetContainer.tsx
+++ b/node_modules/@gorhom/bottom-sheet/src/components/bottomSheetContainer/BottomSheetContainer.tsx
@@ -33,7 +33,7 @@ function BottomSheetContainerComponent({
         overflow: detached ? 'visible' : 'hidden',
       },
     ],
-    [style, detached, topInset, bottomInset]
+    [style, detached, topInset, bottomInset],
   );
   //#endregion
 
@@ -48,16 +48,18 @@ function BottomSheetContainerComponent({
 
       containerRef.current?.measure(
         (_x, _y, _width, _height, _pageX, pageY) => {
+          if (!containerOffset.value) return;
           containerOffset.value = {
             top: pageY ?? 0,
             left: 0,
             right: 0,
             bottom: Math.max(
               0,
-              WINDOW_HEIGHT - ((pageY ?? 0) + height + (StatusBar.currentHeight ?? 0))
+              WINDOW_HEIGHT -
+                ((pageY ?? 0) + height + (StatusBar.currentHeight ?? 0)),
             ),
           };
-        }
+        },
       );
 
       print({
@@ -68,7 +70,7 @@ function BottomSheetContainerComponent({
         },
       });
     },
-    [containerHeight, containerOffset, containerRef]
+    [containerHeight, containerOffset, containerRef],
   );
   //#endregion
 
