diff --git a/node_modules/react-native-reanimated/src/createAnimatedComponent.tsx b/node_modules/react-native-reanimated/src/createAnimatedComponent.tsx
index b1d2b4a..dfc6275 100644
--- a/node_modules/react-native-reanimated/src/createAnimatedComponent.tsx
+++ b/node_modules/react-native-reanimated/src/createAnimatedComponent.tsx
@@ -246,7 +246,7 @@ export default function createAnimatedComponent(
 
       if (componentName?.endsWith('FlashList') && this._component) {
         // @ts-ignore it's FlashList specific: https://github.com/Shopify/flash-list/blob/218f314e63806b4fe926741ef73f8b9cd6ebc7eb/src/FlashList.tsx#L824
-        viewTag = findNodeHandle(this._component.recyclerlistview_unsafe);
+        viewTag = findNodeHandle(this._component.getScrollableNode());
       }
 
       for (const key in this.props) {
