# stashed
diff --git a/node_modules/react-native-iap/src/iap.ts b/node_modules/react-native-iap/src/iap.ts
index ca04197..ed78de0 100644
--- a/node_modules/react-native-iap/src/iap.ts
+++ b/node_modules/react-native-iap/src/iap.ts
@@ -38,7 +38,7 @@ import {
   SubscriptionAmazon,
   SubscriptionAndroid,
   SubscriptionIOS,
-  SubscriptionPlatform,
+  // SubscriptionPlatform,
   SubscriptionPurchase,
 } from './types';
 
@@ -231,6 +231,13 @@ const App = () => {
 ```
 
  */
+
+enum SubscriptionPlatform {
+  android = 'android',
+  amazon = 'amazon',
+  ios = 'ios',
+}
+
 export const getSubscriptions = ({
   skus,
 }: {
diff --git a/node_modules/react-native-iap/src/types/appleSk2.ts b/node_modules/react-native-iap/src/types/appleSk2.ts
index da45bcf..0938170 100644
--- a/node_modules/react-native-iap/src/types/appleSk2.ts
+++ b/node_modules/react-native-iap/src/types/appleSk2.ts
@@ -7,7 +7,7 @@ import type {
   SubscriptionIosPeriod,
 } from '.';
 import type * as Apple from './apple';
-import {SubscriptionPlatform} from '.';
+// import {SubscriptionPlatform} from '.';
 
 export type SubscriptionPeriod = {
   unit: 'day' | 'week' | 'month' | 'year';
@@ -65,6 +65,12 @@ export const productSk2Map = ({
   return prod;
 };
 
+enum SubscriptionPlatform {
+  android = 'android',
+  amazon = 'amazon',
+  ios = 'ios',
+}
+
 export const subscriptionSk2Map = ({
   id,
   description,
