diff --git a/node_modules/react-native/third-party-podspecs/RCT-Folly.podspec b/node_modules/react-native/third-party-podspecs/RCT-Folly.podspec
index cacafb8..c8ea92f 100644
--- a/node_modules/react-native/third-party-podspecs/RCT-Folly.podspec
+++ b/node_modules/react-native/third-party-podspecs/RCT-Folly.podspec
@@ -10,8 +10,8 @@ Pod::Spec.new do |spec|
   spec.homepage = 'https://github.com/facebook/folly'
   spec.summary = 'An open-source C++ library developed and used at Facebook.'
   spec.authors = 'Facebook'
-  spec.source = { :git => 'https://github.com/facebook/folly.git',
-                  :tag => "v#{spec.version}" }
+  spec.source = { :git => 'https://github.com/indaq-cloud/folly.git',
+                  :commit => "660e964e073b65b3a5890819f393daca1e3f8120" }  # commit hash for 2020.01.13.00 + fix on fork
   spec.module_name = 'folly'
   spec.header_mappings_dir = '.'
   spec.dependency 'boost-for-react-native'
