diff --git a/node_modules/react-native-screens/native-stack/index.tsx b/node_modules/react-native-screens/native-stack/index.tsx
index f8b7c7a..b22ed4b 100644
--- a/node_modules/react-native-screens/native-stack/index.tsx
+++ b/node_modules/react-native-screens/native-stack/index.tsx
@@ -13,7 +13,7 @@ export { default as NativeStackView } from './views/NativeStackView';
 /**
  * Types
  */
-export {
+export type {
   NativeStackNavigationOptions,
   NativeStackNavigationProp,
 } from './types';
