diff --git a/node_modules/rn-update-apk/index.js b/node_modules/rn-update-apk/index.js
index 6c8952e..99fd0ab 100644
--- a/node_modules/rn-update-apk/index.js
+++ b/node_modules/rn-update-apk/index.js
@@ -85,7 +85,6 @@ export class UpdateAPK {
     const downloadDestPath = `${RNFS.CachesDirectoryPath}/NewApp.apk`;
 
     let options = this.options.apkOptions ? this.options.apkOptions : {};
-
     const ret = RNFS.downloadFile(
       Object.assign(
         {
@@ -94,6 +93,10 @@ export class UpdateAPK {
           begin,
           progress,
           background: true,
+          headers: {
+            "Cache-Control": "no-transform",
+            "Accept-Encoding": ""
+          },
           progressDivider
         },
         options
