diff --git a/node_modules/vision-camera-ocr/android/src/main/java/com/visioncameraocr/OCRFrameProcessorPlugin.kt b/node_modules/vision-camera-ocr/android/src/main/java/com/visioncameraocr/OCRFrameProcessorPlugin.kt
index 8ae6279..6b47ed6 100644
--- a/node_modules/vision-camera-ocr/android/src/main/java/com/visioncameraocr/OCRFrameProcessorPlugin.kt
+++ b/node_modules/vision-camera-ocr/android/src/main/java/com/visioncameraocr/OCRFrameProcessorPlugin.kt
@@ -111,6 +111,8 @@ class OCRFrameProcessorPlugin: FrameProcessorPlugin("scanOCR") {
             try {
                 val text: Text = Tasks.await<Text>(task)
                 result.putString("text", text.text)
+                result.putInt("width", frame.getWidth())
+                result.putInt("height", frame.getHeight())
                 result.putArray("blocks", getBlockArray(text.textBlocks))
             } catch (e: Exception) {
                 return null
diff --git a/node_modules/vision-camera-ocr/lib/typescript/index.d.ts b/node_modules/vision-camera-ocr/lib/typescript/index.d.ts
index 47f1816..7be2dd6 100644
--- a/node_modules/vision-camera-ocr/lib/typescript/index.d.ts
+++ b/node_modules/vision-camera-ocr/lib/typescript/index.d.ts
@@ -32,6 +32,8 @@ declare type TextBlock = {
 };
 declare type Text = {
     text: string;
+    width: number;
+    height: number;
     blocks: TextBlock[];
 };
 export declare type OCRFrame = {
